src/clj/puppetlabs/puppetserver/shell_utils.clj

```diff
@@ -27,22 +27,25 @@
 (def ExecutionOptions
   {(schema/optional-key :args) [schema/Str]
    (schema/optional-key :env) (schema/maybe {schema/Str schema/Str})
-   (schema/optional-key :in) (schema/maybe InputStream)})
+   (schema/optional-key :in) (schema/maybe InputStream)
+   (schema/optional-key :cwd) (schema/maybe schema/Str)})
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Internal
 
 (def default-execution-options
   {:args []
    :env nil
-   :in nil})
+   :in nil
+   :cwd nil})
 
 (schema/defn ^:always-validate java-exe-options :- ShellUtils$ExecutionOptions
-  [{:keys [env in]} :- ExecutionOptions]
+  [{:keys [env in cwd]} :- ExecutionOptions]
   (let [exe-options (ShellUtils$ExecutionOptions.)]
     (.setStdin exe-options in)
     (when env
       (.setEnv exe-options (ks/mapkeys name env)))
+    (.setWorkingDirectory exe-options cwd)
```

Comment:
Given the schema shows this as an optional key, and an optional result, might be better to use:
```suggestion
    (when env
       (.setWorkingDirectory exe-options cwd))
```
To only execute that statement when the `cwd` is not null?

Reply:
ah sure thing, I was confused because `in` is marked as optional `(schema/optional-key :env)` but is set unconditionally `(.setStdin exe-options in)`